rpm/centos7/common/rke2-common.spec

```diff
@@ -38,5 +40,10 @@ install -m 755 -d %{buildroot}%{_localstatedir}/run/k3s
 %{_sysconfdir}/rancher/rke2/
 %{_sharedstatedir}/rancher/rke2
 %{_localstatedir}/run/k3s
+%{_bindir}/rke2-killall.sh
+%{_bindir}/rke2-uninstall.sh
 
 %changelog
+
+%preun
+INSTALL_RKE2_ROOT=%{_prefix} /usr/bin/rke2-uninstall.sh
```

Comment:
Note that the preun/postun scripts are also run on upgrade:
https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_syntax

> When scriptlets are called, they will be supplied with an argument. This argument, accessed via $1 (for shell scripts) is the number of packages of this name which will be left on the system when the action completes.

We should only run rke2-uninstall.sh if this argument is 0, otherwise we will uninstall rke2 every time the package is upgraded.

Reply:
Thanks @brandond for pointing this out, handled in another pr https://github.com/rancher/rke2-packaging/pull/13